src/Upgrade/Infrastructure/PackageManager/CommandExecutor/ComposerCommandExecutor.php

```diff
@@ -19,12 +20,12 @@ class ComposerCommandExecutor implements ComposerCommandExecutorInterface
     /**
      * @var string
      */
-    protected const REQUIRE_COMMAND_NAME = 'composer require';
+    protected const REQUIRE_COMMAND_NAME = 'COMPOSER_PROCESS_TIMEOUT=0 composer require';
 
     /**
      * @var string
      */
-    protected const UPDATE_COMMAND_NAME = 'composer update';
+    protected const UPDATE_COMMAND_NAME = 'COMPOSER_PROCESS_TIMEOUT=0 composer update';
```

Comment:
I would personally prefer to create a new constant, e.g. `COMPOSER_TIMEOUT` or `COMPOSER_PROCESS_TIMEOUT`, define it as 600 or 900, and extend the places that call composer